src/signalcorr.jl

```diff
@@ -139,36 +134,27 @@ where each column in the result will correspond to a column in `x`.
 The output is normalized by the variance of `x`, i.e. so that the lag 0
 autocorrelation is 1. See [`autocov!`](@ref) for the unnormalized form.
 """
-function autocor!(r::RealVector, x::RealVector, lags::IntegerVector; demean::Bool=true)
-    lx = length(x)
-    m = length(lags)
-    length(r) == m || throw(DimensionMismatch())
-    check_lags(lx, lags)
-
-    T = typeof(zero(eltype(x)) / 1)
-    z::Vector{T} = demean ? x .- mean(x) : x
-    zz = dot(z, z)
-    for k = 1 : m  # foreach lag value
-        r[k] = _autodot(z, lx, lags[k]) / zz
-    end
+function autocor!(
+    r::AbstractVector,
+    x::AbstractVector,
+    lags::AbstractVector{<:Integer},
+    z=zeros(typeof(zero(eltype(x)) / 1), length(x));
+    demean::Bool=true
+    )
+    autocov!(view(r, 1:1), x, 0:0, z; demean)
+    zz = r[1]
```

Comment:
This seems strange. I wonder if we could just update the method in the same way as `autocov!` above?